web/components/vue/InputDropdown.vue

```diff
@@ -0,0 +1,313 @@
+   */
+  placeHolderValue: {
+    type: String,
+    default: "",
+  },
+  /**
+   * Group of values to be selected within the dropdown.
+   */
+  valueGroup: {
+    type: Array<any>,
+    required: true,
+  },
+  /**
+   * Group of names to serve as placeholders for valueGroup, if its length does not match the
+   * valueGroup's the value will be swapped for the valueGroup's values capitalized.
+   */
+  valueNames: {
+    type: Array<String>,
+    default: [],
+  },
+  /**
+   * Value selected to be displayed in the dropdown's field, the actual value being given through
+   * methods
+   */
+  selectedValue: {
+    type: String,
+    default: "",
+  },
+  /**
+   * Additional function to be called whenever an option is selected
+   */
+  onSelect: {
+    type: Function,
+  },
+  /**
+   * Indicates whether the choice made is erroneous
+   */
+  isError: {
+    type: Boolean,
+    default: false,
+  },
+  /**
+   * Controls whether the dropdown is open by default
+   */
+  isOpen: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+// Setting ref values
+let isDropdownOpen = ref(props.isOpen);
+let selectedValue = ref(props.placeHolderValue);
+let selectedValueIndex = ref(props.valueGroup.indexOf(props.selectedValue));
+let hasBeenOpenedViaClick = ref(false);
+
+// Providing the displayed strings using either the valueNames provided or the capitalized-string version of the valueGroup's contents
+let defaultValueNames: Array<string> =
+  props.valueNames.length > 0 && props.valueGroup.length
+    ? props.valueNames
+    : props.valueGroup.map((x) => x.toString()[0].toUpperCase() + x.toString().slice(1));
+
+// Emitter, allows to pass the selected option's value to parent components
+const emit = defineEmits<{ emittedSelectedInput: [value: any] }>();
+
+// Called upon selecting a value, changes the value emitted and calls the onSelect function if it exists
+const onSelectValue = (value: String) => {
+  if (props.onSelect) {
+    props.onSelect();
+  }
+  selectedValue.value = value as string;
+  emit("emittedSelectedInput", value);
+  defineSelectedElementIndex(value as string);
+};
+
+// Opens or closes the dropdown
+const onToggleOpen = () => {
+  isDropdownOpen.value = !isDropdownOpen.value;
+  hasBeenOpenedViaClick.value = !hasBeenOpenedViaClick.value;
+};
+
+// Opens or closes the dropdown if the user clicks outside of the component
+const onClickOutside = () => {
+  if (hasBeenOpenedViaClick.value) {
+    hasBeenOpenedViaClick.value = !hasBeenOpenedViaClick.value;
+    isDropdownOpen.value = !isDropdownOpen.value;
+  }
+};
+
+// Changes the selected element's index to help with display
+const defineSelectedElementIndex = (value: string) => {
+  selectedValueIndex.value = props.valueGroup.indexOf(value);
+};
+
+const classList = [
+  "input__border-background",
+  "input__container-flex",
+  "input__label-style",
+  "dropdown",
+];
+
+const isBrowserMozilla = new RegExp("Firefox").test(navigator.userAgent.toString());
```

Comment:
Check de browser agent c'est pas ouf 
https://developer.mozilla.org/en-US/docs/Learn/Tools_and_testing/Cross_browser_testing/Feature_detection#javascript

Reply:
A fix has been made according to this suggestion, thank you for the assistance.